File: Data Structures & Algorithms/missing-number/submission-1.cpp
class Solution
{
public:
    int missingNumber(vector<int> &nums)
    {
        int missing = 0;
        int i = 0;
        sort (nums.begin(), nums.end());
        while (true)
        {
            if (missing == nums[i])
            {
                missing++;
                i++;
            }
            else
            {
                return missing;
            }
        }
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-1.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        var currentTotal = 0;
        for(int i = 0; i < nums.Length; i++)
        {
            var numsI = nums[i];
            if(dict.ContainsKey(numsI))
            {
                dict[numsI]++;
            }
            else
            {
                dict.Add(numsI, 1);
            }
            currentTotal+= dict[numsI];
            if(currentTotal > nums.Length) return true;
        }

        return false;
    }
}